.werft/platform-trigger-artificial-job.sh

```diff
@@ -15,6 +15,40 @@ sudo chown -R gitpod:gitpod /workspace
 git config --global user.name roboquat
 git config --global user.email roboquat@gitpod.io
 
-git checkout -b "platform-artificial-job-$(echo $RANDOM | md5sum | head -c 20; echo;)"
+BRANCH="platform-artificial-job-$(echo $RANDOM | md5sum | head -c 20; echo;)"
+git checkout -b "$BRANCH" | werft log slice "Creating new branch"
 
-werft run github -a with-preview=true
+BUILD_ID=$(werft run github -a with-preview=true)
+job_url="https://werft.gitpod-dev.com/job/${BUILD_ID}"
+echo "Triggered job $job_url" | werft log slice "Triggering new job"
+werft log result -d "build job" url "$job_url"
+
+while true; do
+    job_phase=$(werft job get "${BUILD_ID}" -o json | jq --raw-output '.phase')
```

Comment:
Werft is currently a bit unreliable (see https://github.com/gitpod-io/gitpod/issues/12628#issuecomment-1239429502) so there is a good chance that this will fail every once in a while.

You could temporarily disable "fail on error" and check the exit code manually. Something like this (not tested)

```sh
set +e
while true; do
    job_phase=$(werft job get "${BUILD_ID}" -o json | jq --raw-output '.phase')
    if [[ "$?" != "0" ]]; then
        echo "Failed to get job details. Sleeping and trying again"
        sleep 10
    fi
    // .. the rest of your code
done
set -e
```

This is how workspace [worked around it for now](https://github.com/gitpod-io/gitpod/blob/main/.werft/workspace-run-integration-tests.sh#L113)

Reply:
Ah nice catch

Reply:
Actually I looked it up and while set +e/-e doesn't make any difference in while loops 🤔 
![image](https://user-images.githubusercontent.com/24193764/190415448-7c985816-ad72-4cf8-a757-3aeaf02edf08.png)

Reply:
@ArthurSens I think that only applies to commands executed in the "condition" part of those expressions. E.g. try to run this script

```sh
#!/usr/bin/env bash

set -euo pipefail

function fail {
    if [[ $1 == "yes" ]]; then
        exit 1
    else
        exit 0
    fi
}

while $(fail "yes"); do
    echo "This shouldn't run"
done

while $(fail "no"); do
    echo "Running"
    sleep 1
    fail "yes"
done
```

You'll see

```
gitpod /workspace/empty (main) $ ./test.sh 
Running
```

E.g. that `fail "yes"` does cause the entire script to fail ☺️